src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.cpp

```diff
@@ -0,0 +1,495 @@
+        break;
+
+    case Commands::ProvideICECandidates::Id:
+        CommandHandlerInterface::HandleCommand<Commands::ProvideICECandidates::DecodableType>(
+            ctx, [this](HandlerContext & subCtx, const auto & req) { HandleProvideICECandidates(subCtx, req); });
+        break;
+
+    case Commands::EndSession::Id:
+        CommandHandlerInterface::HandleCommand<Commands::EndSession::DecodableType>(
+            ctx, [this](HandlerContext & subCtx, const auto & req) { HandleEndSession(subCtx, req); });
+        break;
+
+    default:
+        // Mark unrecognized command as UnsupportedCommand
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::UnsupportedCommand);
+        break;
+    }
+}
+
+// Helper functions
+WebRTCSessionStruct * WebRTCTransportProviderServer::FindSession(uint16_t sessionId)
+{
+    for (auto & session : mCurrentSessions)
+    {
+        if (session.id == sessionId)
+        {
+            return &session;
+        }
+    }
+
+    return nullptr;
+}
+
+void WebRTCTransportProviderServer::AddOrUpdateSession(const WebRTCSessionStruct & session)
+{
+    bool found = false;
+
+    // Search for a session with the same ID and update if found
+    for (auto & existing : mCurrentSessions)
+    {
+        if (existing.id == session.id)
+        {
+            existing = session;
+            found    = true;
+            break;
+        }
+    }
+
+    // If not found, add a new entry
+    if (!found)
+    {
+        mCurrentSessions.push_back(session);
+    }
+
+    // Notify the stack that the CurrentSessions attribute has changed
+    MatterReportingAttributeChangeCallback(AttributeAccessInterface::GetEndpointId().Value(), WebRTCTransportProvider::Id,
+                                           WebRTCTransportProvider::Attributes::CurrentSessions::Id);
+}
+
+void WebRTCTransportProviderServer::RemoveSession(uint16_t sessionId)
+{
+    size_t originalSize = mCurrentSessions.size();
+
+    // Erase-Remove idiom
+    mCurrentSessions.erase(std::remove_if(mCurrentSessions.begin(), mCurrentSessions.end(),
+                                          [sessionId](const WebRTCSessionStruct & s) { return s.id == sessionId; }),
+                           mCurrentSessions.end());
+
+    // If a session was removed, the size will be smaller.
+    if (mCurrentSessions.size() < originalSize)
+    {
+        // Notify the stack that the CurrentSessions attribute has changed.
+        MatterReportingAttributeChangeCallback(AttributeAccessInterface::GetEndpointId().Value(), WebRTCTransportProvider::Id,
+                                               WebRTCTransportProvider::Attributes::CurrentSessions::Id);
+    }
+}
+
+uint16_t WebRTCTransportProviderServer::GenerateSessionID()
+{
+    static uint16_t lastSessionId = 0;
+    uint16_t candidateId          = lastSessionId++;
+
+    // Handle wrap-around per spec
+    if (lastSessionId > 65534)
```

Comment:
magic number: update it with a constant. Also add `{}` for the if here and below (also update magic number there)